### PR TITLE
Remove two clients of MaterialList

### PR DESCRIPTION
**Pull request by abarth**
Clients should use ListView instead.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: examples/flutter_gallery/lib/demo/list_demo.dart [hunk 1]
```diff
@@ -152,7 +152,7 @@ class ListDemoState extends State<ListDemo> {
   @override
   Widget build(BuildContext context) {
     final String layoutText = _dense ? " \u2013 Dense" : "";
-    String  itemTypeText;
+    String itemTypeText;
     switch(_itemType) {
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/list_demo.dart]: trivial style nit while you're here: space after switch

**Comment by abarth** [comment]
Did you want to LGTM this patch?

**Comment by Hixie** [comment]
Sorry, I was writing a comment then got pulled away.

The comment was that to maintain strict parity with what we had before, we really should include an itemExtent on the replacements. But the problem is, do we just hard-code those? Put a constant somewhere? When I looked into doing this the other day, I found that the constants for MaterialList didn't match the constants for ListItem, which seems like an issue.

Anyway, LGTM.

**Comment by abarth** [comment]
I'm not too worried about it for these examples given that they're examples and demos.  We'll need to cross that bridge when we go to remove material list.

**Comment by Hixie** [comment]
Well, our examples and demos are what people are going to copy. So in a way, they're also our customers' code. :-)

**Pull request merged**